### PR TITLE
Fix for Dolby Vision hardware screenshots

### PR DESCRIPTION
**Pull request by thebombzen**
Currently, dolby vision screenshots are not possible to get right with `screenshot-sw=yes` because ITP isn't in H.273 yet, but this fixes the dolby color issue for hardware screenshots.

Closes: #11360.

### CODE REVIEW

**Comment by kasper93** [comment]
Might want to merge https://github.com/mpv-player/mpv/pull/11200 first for proper st428 support.

File: video/csputils.h [hunk 1]
```diff
@@ -67,6 +67,10 @@ enum mp_csp_prim {
     MP_CSP_PRIM_DISPLAY_P3,
     MP_CSP_PRIM_V_GAMUT,
     MP_CSP_PRIM_S_GAMUT,
+    MP_CSP_PRIM_EBU_3213,
+    MP_CSP_PRIM_FILM_C,
+    MP_CSP_PRIM_ACES_AP0,
+    MP_CSP_PRIM_ACES_AP1,
```
Comment (kasper93) [inline video/csputils.h:73]: Adding new enums require more changes if few places to handle them.

File: video/csputils.h [hunk 2]
```diff
@@ -89,6 +93,7 @@ enum mp_csp_trc {
     MP_CSP_TRC_V_LOG,
     MP_CSP_TRC_S_LOG1,
     MP_CSP_TRC_S_LOG2,
+    MP_CSP_TRC_ST428,
```
Comment (kasper93) [inline video/csputils.h:96]: This one is added by https://github.com/mpv-player/mpv/pull/11200 so you don't have to duplicate work.

File: video/csputils.h (hunk 1 above)
Comment (thebombzen) [reply video/csputils.h:73]: Ah, you're right, looking at #11200. Thanks for the heads up

File: video/out/vo_gpu_next.c [hunk 3]
```diff
@@ -1247,21 +1247,11 @@ static void video_screenshot(struct vo *vo, struct voctrl_screenshot *args)
...
-        args->res->params.color = mpi->params.color;
-        args->res->params.color.space = MP_CSP_RGB;
-    } else {
-        args->res->params.color.primaries = MP_CSP_PRIM_BT_709;
-        args->res->params.color.gamma = MP_CSP_TRC_SRGB;
-    }
+    args->res->params.color.primaries = mp_prim_from_pl(target.color.primaries);
+    args->res->params.color.gamma = mp_trc_from_pl(target.color.transfer);
+    args->res->params.color.levels = mp_levels_from_pl(target.repr.levels);
+    args->res->params.color.sig_peak = target.color.sig_peak;
```
Comment (haasn) [inline video/out/vo_gpu_next.c:1253]: This is mapping the wrong field. `target.color.sig_peak` is deprecated and will be removed soon. Should be `target.color.hdr.max_luma / MP_REF_WHITE`

**Review by haasn — APPROVED** [review]
Arguably, these new additions should be documented (in manpage and `interface-changes`). But I don't care too strongly as they;re not all that useful outside of libplacebo compatibility

**Comment by thebombzen** [comment]
In that case I'll go ahead and merge it.

**Pull request merged**